### PR TITLE
Use hooks instead of components

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it cleans code a bit, keeping things tidy.
Converts utility components like `<Navigator />` or `<Paginator />` to hooks, fixes few bugs and simplifies code where I had to touch things anyway.

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/categories/views/CategoryCreate.tsx [hunk 1]
```diff
@@ -16,67 +16,58 @@ interface CategoryCreateViewProps {
...
+      navigate(categoryUrl(data.categoryCreate.category.id));
+    }
+  };
+  return (
+    <TypedCategoryCreateMutation onCompleted={handleSuccess}>
+      {(createCategory, createCategoryResult) => {
+        const formTransitionState = getMutationState(
+          createCategoryResult.called,
+          createCategoryResult.loading,
+          maybe(() => createCategoryResult.data.categoryCreate.errors)
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/categories/views/CategoryCreate.tsx]: To var and bellow `errors || []`

File: saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx [hunk 2]
```diff
@@ -43,89 +43,89 @@ const TextFieldWithChoice = withStyles(styles, {
...
+                open: menuOpen,
+                actions: { open: openMenu, close: closeMenu }
+              }) => {
+                const handleSelect = value => {
+                  onChange({
+                    target: {
+                      name: ChoiceProps.name,
+                      value
+                    }
+                  } as any);
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx]: 👮

File: saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx [hunk 3]
```diff
@@ -43,89 +43,89 @@ const TextFieldWithChoice = withStyles(styles, {
...
-        />
-      )}
-    </Anchor>
-  )
+                return (
+                  <>
+                    <div
+                      className={classes.adornment}
+                      ref={anchor}
+                      onClick={!menuOpen ? openMenu : undefined}
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx]: `!menuOpen && openMenu` won't have the same effect?

File: saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx [hunk 4]
```diff
@@ -43,89 +43,89 @@ const TextFieldWithChoice = withStyles(styles, {
...
+                          }}
+                        >
+                          <Paper>
+                            <ClickAwayListener
+                              onClickAway={closeMenu}
+                              mouseEvent="onClick"
+                            >
+                              <Menu>
+                                {ChoiceProps.values.map(choice => (
+                                  <MenuItem
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx]: Don't U have errors for missing `key` prop?

File: saleor/static/dashboard-next/orders/views/OrderDetails/OrderDetailsMessages.tsx [hunk 5]
```diff
@@ -24,309 +24,291 @@ import { OrderVoid } from "../../types/OrderVoid";
...
+      navigate(orderUrl(data.orderFulfillmentUpdateTracking.order.id), true);
+    } else {
+      pushMessage({
+        text: i18n.t("Could not update fulfillment", {
+          context: "notification"
+        })
+      });
+    }
+  };
+  const handleDraftFinalize = (data: OrderDraftFinalize) => {
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/orders/views/OrderDetails/OrderDetailsMessages.tsx]: I think at least half of these could reuse functionality, i.e. accept result from `OrderDraftFinalize` and others which would be `{errors, order}` and a message. 🙄

File: saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx (hunk 2 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx]: Synthetic events 🤷‍♂️

File: saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx (hunk 3 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx]: Probably, didn't look at it. This is only indent change 🤷‍♂️

File: saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx (hunk 4 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx]: Didn't notice any, but yeah it should have key 🙃

File: saleor/static/dashboard-next/orders/views/OrderDetails/index.tsx [hunk 6]
```diff
@@ -72,606 +73,601 @@ interface OrderDetailsProps {
...
+                  </OrderOperations>
+                )}
+              </OrderDetailsMessages>
+            )}
+          </UserSearchProvider>
+        );
+      }}
+    </TypedOrderDetailsQuery>
+  );
+};
```
[the remark this answers is not in the document]
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/orders/views/OrderDetails/index.tsx]: Indent change 🤷‍♂️

File: saleor/static/dashboard-next/orders/views/OrderDetails/OrderDetailsMessages.tsx (hunk 5 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/orders/views/OrderDetails/OrderDetailsMessages.tsx]: There are plans for some refactoring here, but not in this PR.

File: saleor/static/dashboard-next/categories/views/CategoryCreate.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/categories/views/CategoryCreate.tsx]: Good point.

**Pull request merged**